wrappers/golang/build.sh

```diff
@@ -2,32 +2,59 @@
 
 G2_DEFINED=OFF
 ECNTT_DEFINED=OFF
+CUDA_COMPILER_PATH=/usr/local/cuda/bin/nvcc
+DEVMODE=OFF
 
-if [[ $2 == "ON" ]]
-then
-  G2_DEFINED=ON
-fi
-
-if [[ $3 ]]
-then
-  ECNTT_DEFINED=ON
-fi
-
-BUILD_DIR=$(realpath "$PWD/../../icicle/build")
 SUPPORTED_CURVES=("bn254" "bls12_377" "bls12_381" "bw6_761")
 
+# Parse arguments
+# Handle curve
 if [[ $1 == "all" ]]
 then
   BUILD_CURVES=("${SUPPORTED_CURVES[@]}")
 else
   BUILD_CURVES=( $1 )
 fi
 
+shift # skip curve argument
+for arg in "$@"
+do
+    arg_lower=$(echo "$arg" | tr '[:upper:]' '[:lower:]')
+    case "$arg_lower" in
+        -cuda_version=*)
+            cuda_version=$(echo "$arg" | cut -d'=' -f2)
+            CUDA_COMPILER_PATH=/usr/local/cuda-$cuda_version/bin/nvcc
+            ;;
+        -ecntt*)
+            # ECNTT_DEFINED=$(echo "$arg" | cut -d'=' -f2)
+            ECNTT_DEFINED=ON
+            ;;
+        -g2*)
+            # G2_DEFINED=$(echo "$arg" | cut -d'=' -f2)
+            G2_DEFINED=ON
+            ;;
+        -devmode*)
```

Comment:
We want specific flags and not to match a flag with a prefix

```suggestion
        -ecntt)
            # ECNTT_DEFINED=$(echo "$arg" | cut -d'=' -f2)
            ECNTT_DEFINED=ON
            ;;
        -g2)
            # G2_DEFINED=$(echo "$arg" | cut -d'=' -f2)
            G2_DEFINED=ON
            ;;
        -devmode)
```